src/hammer-vlsi/synthesis/vivado/par.py

```diff
@@ -0,0 +1,34 @@
+from typing import List
+
+import os
+
+from hammer_vlsi import HammerPlaceAndRouteTool, HammerToolStep
+
+from .core import VivadoCommon
```

Comment:
FYI without `__init__.py` the default HammerDriver will get confused when it tries to import this plugin.

We probably need to move `core.py` to something like `src/hammer-vlsi/common/vivado_core.py` and move synth.py and par.py to `src/hammer-vlsi/synthesis/vivado/__init__.py` and `src/hammer-vlsi/par/vivado/__init__.py` respectively.

Reply:
The problem I have is that the current hammer source directory structure is really weird; it seems like you have many disjoints python modules/packages lying all over the place, with no clear hierarchy between them.

For example, src/hammer-vlsi/common is not a python package (no __init__.py), so is the synthesis folder and par folder, so organization wise they are just three disjoint folders that each contains some python stuffs.  with the suggestion you have then the synthesis/place-and-route files need to magically know the existence and path to the vivado_core.py file, and subclass a class defined there.  I'm not so comfortable with that approach.

My current solution now is to have a vivado python package living in the synthesis folder (although admittedly that's not a good place to be in anymore), with a core module, synth module, and par module, so the classes are tightly coupled together.  To import the synthesis plugin, you set the synthesis path to "src/hammer-vlsi/synthesis", and the tool name to "vivado.synth".  To import the P&R plugin, you use the same tool path, but change the name to "vivado.par".  importlib is able to resolve it properly.

Eric

Reply:
Yeah I agree that plugin common files don't have a good structure and just filed #198 to deal with this. I do think in the meantime it would be nice if we could follow the convention for importing hammer plugins so ideally we would put the par portion of vivado to `src/hammer-vlsi/par/vivado`.

If it's difficult I can work on #198 first if you'd like?